tensorflow_lite_support/cc/task/vision/landmark_detector.cc

```diff
@@ -0,0 +1,154 @@
+
+  // Copy options to ensure the ExternalFile outlives the constructed object.
+  auto options_copy = absl::make_unique<LandmarkDetectorOptions>(options);
+
+  std::unique_ptr<LandmarkDetector> landmark_detector;
+  if (options_copy->base_options().has_model_file()) {
+    ASSIGN_OR_RETURN(landmark_detector,
+                     TaskAPIFactory::CreateFromBaseOptions<LandmarkDetector>(
+                         &options_copy->base_options()));
+  } else {
+    // Should never happen because of SanityCheckOptions.
+    return CreateStatusWithPayload(
+        StatusCode::kInvalidArgument,
+        absl::StrFormat("Expected exactly one `base_options.model_file` "
+                        "to be provided, found 0."),
+        TfLiteSupportStatus::kInvalidArgumentError);
+  }
+
+  RETURN_IF_ERROR(landmark_detector->Init(std::move(options_copy)));
+
+  return landmark_detector;
+}
+
+/* static */
+absl::Status LandmarkDetector::SanityCheckOptions(
+    const LandmarkDetectorOptions& options) {
+  // Nothing to check
+  return absl::OkStatus();
+}
+
+absl::Status LandmarkDetector::Init(
+    std::unique_ptr<LandmarkDetectorOptions> options) {
+  // Set options.
+  options_ = std::move(options);
+
+  // Perform pre-initialization actions (by default, sets the process engine for
+  // image pre-processing to kLibyuv as a sane default).
+  RETURN_IF_ERROR(PreInit());
+
+  // Sanity check and set inputs and outputs.
+  RETURN_IF_ERROR(CheckAndSetInputs());
+
+  return absl::OkStatus();
+}
+
+absl::Status LandmarkDetector::PreInit() {
+  SetProcessEngine(FrameBufferUtils::ProcessEngine::kLibyuv);
+  return absl::OkStatus();
+}
+
+StatusOr<LandmarkResult> LandmarkDetector::Detect(
+    const FrameBuffer& frame_buffer) {
+  BoundingBox roi;
+  roi.set_width(frame_buffer.dimension().width);
+  roi.set_height(frame_buffer.dimension().height);
+  return Detect(frame_buffer, roi);
+}
+
+StatusOr<LandmarkResult> LandmarkDetector::Detect(
+    const FrameBuffer& frame_buffer, const BoundingBox& roi) {
+  return InferWithFallback(frame_buffer, roi);
+}
+
+StatusOr<LandmarkResult> LandmarkDetector::Postprocess(
+    const std::vector<const TfLiteTensor*>& output_tensors,
+    const FrameBuffer& /*frame_buffer*/, const BoundingBox& /*roi*/) {
+  if (output_tensors.size() != 1) {
+    return CreateStatusWithPayload(
+        StatusCode::kInternal,
+        absl::StrFormat("Expected 1 output tensors, found %d",
+                        output_tensors.size()));
+  }
+
+  const TfLiteTensor* output_tensor = output_tensors[0];
+
+  const float* outputs = AssertAndReturnTypedTensor<float>(output_tensor);
+	
+  LandmarkResult result;
+
+	for(int i =0 ; i<num_keypoints ; ++i){
+
+    Landmark* landmarks = result.add_landmarks();
+
+		landmarks->set_key_y(outputs[3*i+0]);
+		landmarks->set_key_x(outputs[3*i+1]);
+		landmarks->set_score(outputs[3*i+2]);
+
+  }
+  
+  return result;
+}
+
+
+}  // namespace vision
+}  // namespace task
+}  // namespace tflite
```

Comment:
Every file needs to have a new line at the end. This is the google coding convention. I think clang-format should also help you auto-format in this way.